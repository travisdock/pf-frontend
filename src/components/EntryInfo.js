import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button className="delete" onClick={this.props.askIfSure}>Delete Entry</button>
                <button className="update" onClick={this.props.openUpdateModal}>Update Entry</button>
                <button className="copy" onClick={() => this.props.copyEntry(this.props)}>Copy Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;