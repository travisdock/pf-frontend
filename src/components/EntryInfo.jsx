import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button onClick={this.props.askIfSure}>Delete Entry</button>
                <button onClick={this.props.openUpdateModal}>Update Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;