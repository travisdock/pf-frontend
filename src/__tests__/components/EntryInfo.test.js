import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import EntryInfo from '../../components/EntryInfo';
import { jsxOpeningElement, jsxEmptyExpression } from '@babel/types';

const openProps = {
    open: "info",
    closeModal: jest.fn(),
    category_name: "Test Category",
    date: "12/12/12",
    amount: "123.12",
    notes: "These are notes",
    askIfSure: jest.fn(),
    openUpdateModal: jest.fn(),
    copyEntry: jest.fn()
}

it('renders a snapshot', () => {
    const component = renderer.create(<EntryInfo {...openProps} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

describe('correct functions fire on button clicks', () => {
    test('handleDelete fires on button click', () => {
        const component = shallow(<EntryInfo {...openProps} />);
        component.find('button.delete').simulate('click')
        expect(openProps.askIfSure).toHaveBeenCalled();
    });
    test('closeModal fires on button click', () => {
        const component = shallow(<EntryInfo {...openProps} />);
        component.find('button.update').simulate('click')
        expect(openProps.openUpdateModal).toHaveBeenCalled();
    });
    test('closeModal fires on button click', () => {
        const component = shallow(<EntryInfo {...openProps} />);
        component.find('button.copy').simulate('click')
        expect(openProps.copyEntry).toHaveBeenCalled();
    });
});

