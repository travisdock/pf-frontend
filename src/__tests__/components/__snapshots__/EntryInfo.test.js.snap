// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "left": "0px",
        "position": "absolute",
        "top": "0px",
      }
    }
  />,
  <div
    className="popup-overlay "
    onClick={[Function]}
    style={
      Object {
        "background": "rgba(0, 0, 0,0.5)",
        "bottom": "0",
        "display": "flex",
        "left": "0",
        "position": "fixed",
        "right": "0",
        "top": "0",
        "zIndex": "999",
      }
    }
  >
    <div
      className="popup-content "
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(255, 255, 255)",
          "border": "1px solid rgb(187, 187, 187)",
          "margin": "auto",
          "padding": "5px",
          "position": "relative",
          "width": "50%",
        }
      }
    >
      <div
        className="table-popup"
      >
        <p>
          Amount: 
          $123.12
        </p>
        <p>
          Category: 
          Test Category
        </p>
        <p>
          Date: 
          12/12/12
        </p>
        <p>
          Notes: 
          These are notes
        </p>
        <button
          className="delete"
          onClick={[MockFunction]}
        >
          Delete Entry
        </button>
        <button
          className="update"
          onClick={[MockFunction]}
        >
          Update Entry
        </button>
        <button
          className="copy"
          onClick={[Function]}
        >
          Copy Entry
        </button>
      </div>
    </div>
  </div>,
]
`;
